analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.Implementation.cs

```diff
@@ -20,95 +20,76 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using Microsoft.CodeAnalysis;
 
 namespace SonarAnalyzer.Helpers
 {
-    internal abstract partial class KnownType
+    [DebuggerDisplay("{FullName}, Array: {IsArray}, Parameters: {string.Join(\",\", genericParameters)}", Name = "{TypeName}")]
```

Comment:
Isn't this too hard to read? It dumps the technical internal state, but itsn't  very close to the more dev-friendly form of `System.Action<T>[]`?

It's meant to be in debug-only, so I'd be fine with `ToString()` and building it on the fly.

Reply:
I wanted to keep things simple. We also want to be language-independent on KnownType and generating C# syntax for VB might be confusing.

Reply:
> I wanted to keep things simple. We also want to be language-independent on KnownType and generating C# syntax for VB might be confusing.

In this case I would say that that would not be an issue; the VB.NET way of displaying types is not nice (understatement), and despite the context, reading a known type the C# way is - imho - the preferred way. Just my 2cts.

Reply:
We agreed to improve it in next PR. C# is fine here, it's debug-only purposes